docs/tutorials/02_nocturne_concepts.md

```diff
@@ -0,0 +1,323 @@
+```
+
+```python
+# Number of road lines
+len(scenario.road_lines())
+```
+
+```python
+scenario.getVehicles()[:5]
+```
+
+```python
+# No cyclists in this scene
+scenario.getCyclists()
+```
+
+```python
+# Select all moving vehicles that move 
+moving_vehicles = [obj for obj in scenario.getVehicles() if obj in objects_that_moved]
+
+print(f'Found {len(moving_vehicles)} moving vehicles in scene: {[vehicle.getID() for vehicle in moving_vehicles]}')
+```
+
+#### Ego state
+
+The **ego state** is an array with features that describe the current vehicle. This array holds the following information: 
+- 0: length of ego vehicle
+- 1: width of ego vehicle
+- 2: speed of ego vehicle
+- 3: distance to the goal position of ego vehicle
+- 4: angle to the goal (target azimuth) 
+- 5: desired heading at goal position
+- 6: desired speed at goal position
+- 7: current acceleration
+- 8: current steering position
+- 9: current head angle
+
+```python
+# Select an arbitrary vehicle
+ego_vehicle = moving_vehicles[0]
+
+print(f'Selected vehicle # {ego_vehicle.getID()}')
+
+# Get the state for ego vehicle
+scenario.ego_state(ego_vehicle)
+```
+
+#### Visible state
+
+We use the ego vehicle state, together with a view distance (how far the vehicle can see) and a view angle to construct the **visible state**. The figure below shows this procedure for a simplified traffic scene. 
+
+Calling `scenario.visible_state()` returns a dictionary with four matrices:
+- `stop_signs`: The visible stop signs 
+- `traffic_lights`: The states for the traffic lights from the perspective of the ego driver(red, yellow, green).
+- `road_points`: The observable road points (static elements in the scene).
+- `objects`: The observable road objects (vehicles, pedestrians and cyclists).
+
+<figure>
+<center>
+<img src='https://drive.google.com/uc?id=1fG43NvPCzaimmW99asRdB73qY-F4u-q0' width='700'/>
+<figcaption>To investigate coordination under partial observability, agents in Nocturne can only see an obstructed view of their environment. In this simplified traffic scene, we construct the state for the red ego driver. Note that Nocturne assumes that stop signs can be viewed, even if they are behind another driver. </figcaption></center>
+</figure>
+
+\begin{align*}
+\end{align*}
+
+<figure>
+<center>
+<img src='https://drive.google.com/uc?id=1egNkFArE-n4cp6KbeoQyWeePiQ28jYYE' width='300'/>
+<figcaption>The same scene, this time showing the view of the yellow car.</figcaption></center>
+</figure>
+
+
+The shape of the visible state is a function of the maximum number of visible objects defined at initialization (traffic lights, stop signs, road objects, and road points) and whether we add padding. If `padding = True`, an array is of size `(max visible objects, # features)` is always constructed, even if there are no visible objects. Otherwise, if `padding = False` new entries are only created when objects are visible.
```

Comment:
I'd add an explanation of what the padding does. For example, if we have less than max_visible_objects then...